google/resource_container_node_pool.go

```diff
@@ -415,6 +460,37 @@ func nodePoolUpdate(d *schema.ResourceData, meta interface{}, clusterName, prefi
 		}
 	}
 
+	if d.HasChange(prefix + "management") {
+
+		management := &container.NodeManagement{}
+		if v, ok := d.GetOk(prefix + "management"); ok {
+			managementConfig := v.([]interface{})[0].(map[string]interface{})
+			management.AutoRepair = managementConfig["auto_repair"].(bool)
+			management.AutoUpgrade = managementConfig["auto_upgrade"].(bool)
+		}
+		req := &container.SetNodePoolManagementRequest{
+			Management: management,
+		}
+		op, err := config.clientContainer.Projects.Zones.Clusters.NodePools.SetManagement(
+			project, zone, clusterName, npName, req).Do()
+
+		if err != nil {
+			return err
+		}
+
+		// Wait until it's updated
+		waitErr := containerOperationWait(config, op, project, zone, "updating GKE node pool management", timeoutInMinutes, 2)
+		if waitErr != nil {
+			return waitErr
+		}
+
+		log.Printf("[INFO] Updated management in Node Pool %s", d.Id())
```

Comment:
I'd use `npName` instead of `d.Id()` since this function gets called for `google_container_node_pool` resources as well as `google_container_cluster` ones.

Reply:
Done.